app/grails-app/services/org/grails/im/plugins/slack/SlackService.groovy

```diff
@@ -75,6 +74,9 @@ class SlackService implements GrailsConfigurationAware {
                 attachments: attachments,
             ]
 
+            String urlParams = params.keySet().collect { "${it}={${it}}" }.join('&')
```

Comment:
Extract this into a method 

String generateUrlParamsStringFromMap(Map m)

and reuse it in both methods. 

So that we don't have this code 

String urlParams = params.keySet().collect { "${it}={${it}}" }.join('&')

duplicated

Reply:
Done in 9548c50e1ca814524b991df96627f7d64fd5d43e